.github/workflows/pull_request.yml

```diff
@@ -7,24 +7,21 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  update_build_test:
-    name: Update, Build, Test, and compute Test Coverage
+  pipeline:
+    name: Pipeline - code coverage + dependency check
     runs-on: ubuntu-latest
     
     steps:
       - uses: actions/checkout@v3
 
-      # install action-rs toolchain 
-      # deprecated, but best tool for usage with clippy in PR context and only way to calculate test coverage
-      - uses: actions-rs/toolchain@v1
+      - name: Setup dtolnay/rust-toolchain 
+        uses: dtolnay/rust-toolchain@stable
         with:
-          profile: minimal
           toolchain: stable
-          override: true
-          component: clippy, rustfmt
-      
-      # load cargo cache to reduce compilation time
-      - name: Set up cargo cache
+          components: clippy, rustfmt
+
+      # load project cache to reduce compilation time
+      - name: Setup project cache

```

Comment:
The `checkout` operation does not take up any amount of time s.t. it would be worth considering caching an additional folder. If we would need to, we could also add it to the list of folders in the customly configured cache, but I don't see a reason for that.

Reply:
Sounds reasonable